process/transaction/baseProcess.go

```diff
@@ -218,30 +221,54 @@ func (txProc *baseTxProcessor) VerifyTransaction(tx *transaction.Transaction) er
 	return txProc.checkTxValues(tx, senderAccount, receiverAccount, false)
 }
 
+// Setting a guardian is allowed with regular transactions on a guarded account
+// but in this case is set with the default epochs delay
+func checkOperationAllowedToBypassGuardian(txData []byte) error {
+	if process.IsSetGuardianCall(txData) {
+		return nil
+	}
+
+	return fmt.Errorf("%w, not allowed to bypass guardian", process.ErrTransactionNotExecutable)
+}
+
+func (txProc *baseTxProcessor) checkGuardedAccountUnguardedTxPermission(txData []byte, account state.UserAccountHandler) error {
+	err := checkOperationAllowedToBypassGuardian(txData)
+	if err != nil {
+		return err
+	}
+
+	// block non guarded setGuardian Txs if there is a pending guardian
+	hasPendingGuardian := txProc.guardianChecker.HasPendingGuardian(account)
+	if process.IsSetGuardianCall(txData) && hasPendingGuardian {
```

Comment:
I think the condition `process.IsSetGuardianCall(txData)` is not really necessary, as it is already checked in checkOperationAllowedToBypassGuardian method called in L235. In this point, L242, this condition is always true.